shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -1460,6 +1467,21 @@ private void sendViewportMetricsToFlutter() {
 
     viewportMetrics.devicePixelRatio = getResources().getDisplayMetrics().density;
     viewportMetrics.physicalTouchSlop = ViewConfiguration.get(getContext()).getScaledTouchSlop();
+
+    if (Build.VERSION.SDK_INT >= API_LEVELS.API_35) {
```

Comment:
If the other class changes getCaptionBarInsets then this api check should also change.